### PR TITLE
EVEREST-838 Fix uninstall command

### PR DESCRIPTION
**Pull request by recharte**
**Fix uninstall command**
---
**Problem:**
EVEREST-838

Rename OLM namespace to `everest-olm`
Remove useless `operators` namespace
Uninstall command now deletes everything (if user approves it):
* Delete DBCs (they can have finalizers, should be deleted first)
* Delete BackupStorages (they can have finalizers, should be deleted first)
* Delete DB namespaces
* Delete monitoring namespace
* Delete system namespace
* Delete OLM namespace

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?

**Tests**
- [ ] Is an Integration test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?

### CODE REVIEW

File: pkg/kubernetes/client/customresources/backupstorages.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package customresources provides methods to work with custom everest k8s resources.
+//
+//nolint:dupl
```
Comment (michal-kralik) [inline pkg/kubernetes/client/customresources/backupstorages.go:18]: @recharte why do we continue generating these files in custom resources for every resource when we can use a generic solution?
Each file is ~100 lines and needs to be maintained...

File: pkg/uninstall/uninstall.go [hunk 2]
```diff
@@ -80,40 +81,221 @@ This will uninstall Everest and all monitoring resources deployed by it. All oth
...
+		}
 	}
 
-	return nil
+	// Wait for all namespaces to be deleted, or timeout after 5 minutes.
+	u.l.Infof("Waiting for namespaces to be deleted")
+	timeout := time.After(5 * time.Minute)
+	tick := time.Tick(5 * time.Second)
+	for {
+		select {
```
Comment (michal-kralik) [inline pkg/uninstall/uninstall.go:233]: This is a good place to wait for ctx cancellation

File: pkg/kubernetes/database_cluster_restore.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	"context"
+
+	everestv1alpha1 "github.com/percona/everest-operator/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GetDatabaseClusterRestore returns database cluster restore by name.
+func (k *Kubernetes) GetDatabaseClusterRestore(ctx context.Context, namespace, name string) (*everestv1alpha1.DatabaseClusterRestore, error) {
+	return k.client.GetDatabaseClusterRestore(ctx, namespace, name)
```
Comment (michal-kralik) [inline pkg/kubernetes/database_cluster_restore.go:27]: These functions don't contain any logic and just call `k.client`
Does it make sense to have them in place if they don't provide value?

File: pkg/uninstall/uninstall.go (hunk 2 above)
Comment (recharte) [reply pkg/uninstall/uninstall.go]: refactored

File: pkg/kubernetes/client/client.go [hunk 4]
```diff
@@ -217,13 +218,17 @@ func (c *Client) setup() error {
...
 	}
-	c.dbClusterClient = dbClusterClient
+	c.customClientSet = customClient
 	_, err = c.GetServerVersion()
-	return err
+	if err != nil {
+		return err
+	}
+
+	return nil
```
[the remark this answers is not in the document]
Comment (recharte) [reply pkg/kubernetes/client/client.go:231]: no :see_no_evil:

**Review by oksana-grishchenko — APPROVED** [review]
🚀
I believe the massive amount of the new code is not is not going to be needed anymore with the monorepo. So I'm ok with having it in this release

**Comment by recharte** [comment]
> 🚀 I believe the massive amount of the new code is not is not going to be needed anymore with the monorepo. So I'm ok with having it in this release

Exactly my thought, this was the quickest path forward and is already part of the common code in the monorepo so :shrug:

**Pull request merged**